docs/reference/cryptographic-overview.md

```diff
@@ -0,0 +1,47 @@
+### Signature Verification
+Cloud image is verified by SHA256 checksum. Runner binary is also downloaded by [GitHub Runner Charm](https://github.com/canonical/github-runner-operator) and verified by SHA256 in this charm.
+### User SSH Access
+Sometimes users need to access the VM instance that is running the workload, to establish this connection [Tmate](https://tmate.io/) is used. Tmate uses SSH protocol to secure shell connections between users and the GitHub runner. The connection is secured with rsa keypair and ed25519 fingerprints.
+## Cryptographic Technology Used by the Product
+The following cryptographic technologies are used internally by our product:
+### TLS
+- Communication with GitHub is done via TLS v1.3
+- [The Repo Policy Compliance tool](https://github.com/canonical/repo-policy-compliance) communicates with the [GitHub API](https://docs.github.com/en/rest?apiVersion=2022-11-28) using TLS v1.3.
+- Communication with the OpenStack API uses TLS v1.3.
+### Signature Verification
+Cloud images are verified by SHA256 checksum. Runner binary is also downloaded by [the GitHub Runner Charm](https://github.com/canonical/github-runner-operator) and verified by SHA256 in this charm.
+### User SSH Access
+Tmate secures SSH connections using [OpenSSL](https://www.openssl.org/).
+### RSA
+The communication between these virtual machines and runner is done via SSH and secured by RSA 256 byte keypairs.
+## Cryptographic Technology Exposed to Users
+The following sections describe the cryptographic technologies exposed to the user:
+### Tmate
+- [Tmate](https://tmate.io/) uses RSA (384 byte) and ED25519 (32 byte) fingerprints for connections from users to [Tmate](https://tmate.io/) and from the managers to [Tmate](https://tmate.io/). [OpenSSL](https://www.openssl.org/) is being used by [Tmate](https://tmate.io/) to secure the connection between user/manager to runner.
+### Docker Hub Cache
+[Docker Hub cache](https://github.com/canonical/docker-registry-charm) connection is secured via TLS 1.3 and certified by [Let’s Encrypt](https://letsencrypt.org/).
+### Aproxy
+[Aproxy](https://github.com/canonical/aproxy) supports TLS 1.0 and above except TLS 1.3 Encrypted Client Hello.
+## Packages or Technology Providing Cryptographic Functionality
+The following packages or technologies provide cryptographic functionality:
+- [OpenSSL](https://www.openssl.org/) library is being used.
+- For TLS and HTTPS connections urllib3 python library is used.
+- Default clients in Ubuntu for SSH and TLS are being used.
+- [Python hashlib](https://docs.python.org/3/library/hashlib.html) package is being used for sha256 checksum calculation/verification of runner binary and Cloud Init.
```

Comment:
Nitpick: Does sha256 need to be capitalised?

Reply:
When I search for it online it is always capitalized but still wrong I guess. I see it as `SHA-256` in a lot of places.

Ex.: [Wikipedia](https://en.wikipedia.org/wiki/SHA-2), [SimpliLearn](https://www.simplilearn.com/tutorials/cyber-security-tutorial/sha-256-algorithm)